### PR TITLE
Send extra params to the request payload

### PR DESCRIPTION
**Pull request by croqaz**

### CODE REVIEW

**Comment by kmike** [comment]
Hey! I think it may need 3 changes:

1. move "extra" under autoextract key, it shouldn't be a top-level meta field
2. raise an error if "extra" is not a dictionary or None, instead of skipping it silently
3. add documentation to README

**Comment by croqaz** [comment]
Done. Ready to merge 👍

File: test/test_autoextract.py [hunk 1]
```diff
@@ -105,3 +106,15 @@ def test_timeout():
     out = mw.process_request(req, spider)
     assert out is not None
     assert out.meta['download_timeout'] == 10000
+
+
+def test_meta_extra():
+    config = dict(MW_SETTINGS)
+    mw = _mock_mw(spider, config)
+    meta = dict(AUTOX_META)
+    meta['autoextract']['extra'] = {'some': 'stuff'}
```
Comment (kmike) [inline test/test_autoextract.py]: you're modifying AUTOX_META inplace here, because dict(AUTOX_META) is a shallow copy

Comment (croqaz) [reply test/test_autoextract.py]: Good point :)
I would have noticed in the next text, when failing for weird reasons
Fixing now

**Comment by kmike** [comment]
Thanks @croqaz!

**Pull request merged**